app/src/main/java/org/mozilla/rocket/download/DownloadViewModelFactory.kt

```diff
@@ -0,0 +1,29 @@
+package org.mozilla.rocket.download
+
+import android.arch.lifecycle.ViewModel
+import android.arch.lifecycle.ViewModelProvider
+import org.mozilla.focus.Inject
+
+class DownloadViewModelFactory private constructor(private val repository: DownloadInfoRepository) : ViewModelProvider.NewInstanceFactory() {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
```

Comment:
A really fancy way to do things like this (without Factory) is:
```
inline fun <reified T: ViewModel> ViewModel.create(application: Application, repository: DownloadInfoRepository): T {
        if (this is DownloadIndicatorViewModel) {
            return DownloadIndicatorViewModel(application, repository) as T
        } else if (this is DownloadInfoViewModel) {
            return DownloadInfoViewModel(repository) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class: " + this.javaClass.name)
    }
```
This is what I just read from https://kotlinlang.org/docs/reference/inline-functions.html but I'm actually unsure if this is what we want here.